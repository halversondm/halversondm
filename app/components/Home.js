/**
 * Created by Daniel on 6/26/2016.
 */
"use strict";

import React from "react";
import {Link} from "react-router";

const Home = React.createClass({
  render() {
    return (
      <div>
        <div className="jumbotron">
          <h1>@halversondm</h1>
          <p>Website of Dan Halverson</p>
          <p>
            <Link className="btn btn-primary btn-lg" to="about" role="button">Learn
              more &raquo;</Link>
          </p>
        </div>
        <div className="row">
          <div className="col-md-3">
            <h2 className="text-primary">Auto</h2>
            <p>My favorite cars and motorcycles.</p>
            <p>
              <Link className="btn btn-info" to="auto" role="button">View
                details &raquo;</Link>
            </p>
          </div>
          <div className="col-md-3">
            <h2 className="text-primary">Apps</h2>
            <p>Web Applications that I have written.</p>
            <p>
              <Link className="btn btn-info" to="apps" role="button">View
                details &raquo;</Link>
            </p>
          </div>
          <div className="col-md-3">
            <h2 className="text-primary">Resume</h2>
            <p>Who I've worked for and my education details</p>
            <p>
              <Link className="btn btn-info" to="resume" role="button">View
                details &raquo;</Link>
            </p>
          </div>
          <div className="col-md-3">
            <h2 className="text-primary">Blog</h2>
            <p>My Tech Blog where I opine on ... tech</p>
            <p>
              <Link className="btn btn-info" to="blog" role="button">View
                details &raquo;</Link>
            </p>
          </div>
        </div>
      </div>
    );
  }
});

export default Home;
