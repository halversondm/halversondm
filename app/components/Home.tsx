/**
 * Created by Daniel on 6/26/2016.
 */
import * as React from 'react'
import {Link} from 'react-router-dom'
import {type ReactNode} from 'react'

export class Home extends React.Component<unknown, unknown> {
    render(): ReactNode {
        return (
            <div>
                <div className="jumbotron">
                    <h1>@halversondm</h1>
                    <p>Website of Dan Halverson</p>
                    <p>
                        <Link className="btn btn-primary btn-lg" to="/about" role="button">Learn
                            more &raquo;</Link>
                    </p>
                </div>
                <div className="row">
                    <div className="col-md-3">
                        <h2 className="text-primary">Auto</h2>
                        <p>My favorite cars and motorcycles.</p>
                        <p>
                            <Link className="btn btn-info" to="/auto" role="button">View
                                details &raquo;</Link>
                        </p>
                    </div>
                    <div className="col-md-3">
                        <h2 className="text-primary">Apps</h2>
                        <p>Web Applications that I have written.</p>
                        <p>
                            <Link className="btn btn-info" to="/apps" role="button">View
                                details &raquo;</Link>
                        </p>
                    </div>
                    <div className="col-md-3">
                        <h2 className="text-primary">Resume</h2>
                        <p>Who I&apos;ve worked for and my education details</p>
                        <p>
                            <Link className="btn btn-info" to="/resume" role="button">View
                                details &raquo;</Link>
                        </p>
                    </div>
                    <div className="col-md-3">
                        <h2 className="text-primary">Blog</h2>
                        <p>My Tech Blog where I opine on ... tech</p>
                        <p>
                            <Link className="btn btn-info" to="/blog" role="button">View
                                details &raquo;</Link>
                        </p>
                    </div>
                </div>
            </div>
        )
    }
}
