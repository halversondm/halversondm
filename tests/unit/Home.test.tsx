import React from "react";
import { render } from '@testing-library/react';
import Home from "../../app/components/Home";
import {BrowserRouter} from "react-router-dom";

test("Home component", () => {
    const unit = <Home />;
    const component = render(unit,{wrapper: BrowserRouter});
    const tree = component.container;
    expect(tree.toString()).toMatchSnapshot();
});