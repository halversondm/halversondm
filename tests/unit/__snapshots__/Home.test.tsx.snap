// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home component 1`] = `"<div><div><div class="jumbotron"><h1>@halversondm</h1><p>Website of Dan Halverson</p><p><a class="btn btn-primary btn-lg" role="button" href="/about">Learn more »</a></p></div><div class="row"><div class="col-md-3"><h2 class="text-primary">Auto</h2><p>My favorite cars and motorcycles.</p><p><a class="btn btn-info" role="button" href="/auto">View details »</a></p></div><div class="col-md-3"><h2 class="text-primary">Apps</h2><p>Web Applications that I have written.</p><p><a class="btn btn-info" role="button" href="/apps">View details »</a></p></div><div class="col-md-3"><h2 class="text-primary">Resume</h2><p>Who I've worked for and my education details</p><p><a class="btn btn-info" role="button" href="/resume">View details »</a></p></div><div class="col-md-3"><h2 class="text-primary">Blog</h2><p>My Tech Blog where I opine on ... tech</p><p><a class="btn btn-info" role="button" href="/blog">View details »</a></p></div></div></div></div>"`;
